weblate/addons/gettext.py

```diff
@@ -61,11 +62,26 @@ class GenerateMoAddon(GettextBaseAddon):
     description = _(
         'Automatically generates MO file for every changed PO file.'
     )
+    settings_form = GenerateMoForm
 
     def pre_commit(self, translation, author):
         exporter = MoExporter(translation=translation)
         exporter.add_units(translation.unit_set.all())
         output = translation.get_filename()[:-2] + 'mo'
+
+        if self.instance.configuration['path']:
+            mo_name = os.path.basename(output)
+            if mo_name:
+                mo_path = os.path.dirname(output)
+                if not mo_path:
+                    mo_path = ''
```

Comment:
The `get_filename` returns absolute path, so this should not happen and we want to crash in case dirname doesn't work.

Reply:
If you are sure it's a full path, then I will remove the mo_path check.  The reason I left it in was if it was a relative path and in the root of the component (ie, filemask was just *.po) then there would be no directory path and since the path.join expects strings, I was ensuring it was a string.

Reply:
Yes, I'm sure, it's built here:

https://github.com/WeblateOrg/weblate/blob/98876cd6c770720b5e1106c3a053d7eceb962524/weblate/trans/models/translation.py#L200-L202